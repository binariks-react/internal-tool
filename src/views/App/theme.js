export const getColor = (color, props) => (
  props ? props.theme.colors[color] : ({ theme }) => theme.colors[color]
);

export const getActiveColor = (color, props) => (
  props ? props.theme.colors.activeColors[color] : ({ theme }) => theme.colors.activeColors[color]
);

export const getHoverColor = (color, props) => (
  props ? props.theme.colors.hoverColors[color] : ({ theme }) => theme.colors.hoverColors[color]
);

export const getHoverShadowColor = (color, props) => (
  props ? props.theme.shadows.hover[color] : ({ theme }) => theme.shadows.hover[color]
);

export const getFocusShadowColor = (color, props) => (
  props ? props.theme.shadows.focus[color] : ({ theme }) => theme.shadows.focus[color]
);

export default {
  colors: {
    text: '#5a6169',
    lightGrey: '#e9ecef',
    darkGrey: '#becad6',
    border: '#e1e5eb',
    primary: '#007bff',
    secondary: '#5a6169',
    success: '#17c671',
    danger: '#c4183c',
    warning: '#ffb400',
    info: '#00b8d8',
    dark: '#212529',
    white: '#ffffff',
    royalBlue: '#674eec',
    java: '#1adba2',
    salmon: '#ff4169',
    hoverColors: {
      lightGrey: '#e9ecef',
      primary: '#006fe6',
      secondary: '#4e545b',
      success: '#14af64',
      danger: '#ad1535',
      warning: '#e6a200',
      info: '#00a2bf',
      dark: '#16181b',
      white: '#ffffff',
      royalBlue: '#5337ea',
      java: '#17c491',
      salmon: '#ff2855',
    },
    activeColors: {
      primary: '#0062cc',
      secondary: '#42484e',
      lightGrey: '#FBFBFB',
      success: '#129857',
      danger: '#97122e',
      warning: '#cc9000',
      info: '#008da5',
      dark: '#0a0c0d',
      white: '#007bff',
      royalBlue: '#3f20e7',
      java: '#15ad80',
      salmon: '#ff0e41',
    },
  },
  shadows: {
    hover: {
      primary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,123,255,.25)',
      secondary: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(90,97,105,.25)',
      success: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(23,198,113,.25)',
      danger: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(196,24,60,.25)',
      warning: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,180,0,.25)',
      info: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(0,184,216,.25)',
      dark: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(33,37,41,.25)',
      white: '0 0.125rem 0.625rem rgba(129,142,163,.2), 0 0.0625rem 0.125rem rgba(129,142,164,.3)',
      royalBlue: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(103,78,236,.25)',
      java: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(26,219,162,.25)',
      salmon: '0 5px 15px rgba(0,0,0,.05), 0 4px 10px rgba(255,65,105,.25)',
      inset: 'inset 0 3px 5px rgba(0,0,0,.125)!important',
    },
    focus: {
      primary: '0 0 0 3px rgba(0,123,255,.15),0 3px 15px rgba(0,123,255,.2),0 2px 5px rgba(0,0,0,.1)',
      secondary: '0 0 0 3px rgba(90,97,105,.15),0 3px 15px rgba(90,97,105,.2),0 2px 5px rgba(0,0,0,.1)',
      success: '0 0 0 3px rgba(23,198,113,.15),0 3px 15px rgba(23,198,113,.2),0 2px 5px rgba(0,0,0,.1)',
      danger: '0 0 0 3px rgba(196,24,60,.15),0 3px 15px rgba(196,24,60,.2),0 2px 5px rgba(0,0,0,.1)',
      warning: '0 0 0 3px rgba(255,180,0,.15),0 3px 15px rgba(255,180,0,.2),0 2px 5px rgba(0,0,0,.1)',
      info: '0 0 0 3px rgba(0,184,216,.15),0 3px 15px rgba(0,184,216,.2),0 2px 5px rgba(0,0,0,.1)',
      dark: '0 0 0 3px rgba(33,37,41,.15),0 3px 15px rgba(33,37,41,.2),0 2px 5px rgba(0,0,0,.1)',
      white: '0 0 0 3px rgba(251,251,251,.15),0 3px 15px rgba(251,251,251,.2),0 2px 5px rgba(0,0,0,.1)',
      royalBlue: '0 0 0 3px rgba(103,78,236,.15),0 3px 15px rgba(103,78,236,.2),0 2px 5px rgba(0,0,0,.1)',
      java: '0 0 0 3px rgba(26,219,162,.15),0 3px 15px rgba(26,219,162,.2),0 2px 5px rgba(0,0,0,.1)',
      salmon: '0 0 0 3px rgba(255,65,105,.15),0 3px 15px rgba(255,65,105,.2),0 2px 5px rgba(0,0,0,.1)',
      switch: '0 3px 8px 1px rgba(0,0,0,.2)',
    },
  },
};
