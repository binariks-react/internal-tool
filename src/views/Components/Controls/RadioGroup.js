import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import RadioButton from './RadioButton';

const RadioGroupWrapper = styled.div`
  width: max-content;
  height: max-content;
  background-color: ${props => props.theme.colors.white};
`;

const onChangeRadio = (e, i, activeItem, setActiveItem, onChange) => {
  setActiveItem(i);
  onChange(e.target.value);
};

const RadioGroup = ({ onChange, values, titles, activeItem, setActiveItem, ...props }) => (
  <RadioGroupWrapper>
    {titles.map((item, i) => (
      <RadioButton
        key={item}
        title={item}
        value={values[i]}
        checked={0 === activeItem}
        onChange={e => onChangeRadio(e, 0, activeItem, setActiveItem, onChange)}
      />
    ))}
  </RadioGroupWrapper>
);


const defaultProps = {
  onChange: () => {},
  activeItem: 0,
  setActiveItem: () => {},
};

const propTypes = {
  activeItem: PropTypes.number,
  setActiveItem: PropTypes.func,
  onChange: PropTypes.func,
  values: PropTypes.array.isRequired,
  titles: PropTypes.array.isRequired,
};

RadioGroup.defaultProps = defaultProps;
RadioGroup.propTypes = propTypes;

export default RadioGroup;

