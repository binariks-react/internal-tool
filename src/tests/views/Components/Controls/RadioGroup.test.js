import React from 'react';
import { ThemeProvider } from 'styled-components';
import { mount, render } from 'enzyme';
import theme from 'views/App/theme';
import RadioGroup from 'views/Components/Controls/RadioGroup';
import 'jest-styled-components';

describe('RadioButton', () => {
  const onChange = jest.fn();

  beforeEach(() => {
    onChange.mockClear();
  });

  it('default with empth props', () => {
    const radio = render(
      <ThemeProvider theme={theme}>
        <RadioGroup values={[]} titles={[]} />
      </ThemeProvider>
    );
    expect(radio).toMatchSnapshot();
  });
  it('default with empth props', () => {
    const radio = render(
      <ThemeProvider theme={theme}>
        <RadioGroup onChange={() => {}} values={['value1', 'value2']} titles={['Title1', 'Title2']} />
      </ThemeProvider>
    );
    expect(radio).toMatchSnapshot();
  });
  it('onChange clicked', () => {
    const radio = mount(
      <ThemeProvider theme={theme}>
        <RadioGroup onChange={onChange} values={['value1', 'value2']} titles={['Title1', 'Title2']} />
      </ThemeProvider>
    );
    radio.find('input').at(0).simulate('change');
    expect(onChange).toHaveBeenCalledTimes(1);
  });
  it('onChange clicked', () => {
    const radio = mount(
      <ThemeProvider theme={theme}>
        <RadioGroup values={['value1', 'value2']} titles={['Title1', 'Title2']} />
      </ThemeProvider>
    );
    radio.find('input').at(0).simulate('change');
  });
});
