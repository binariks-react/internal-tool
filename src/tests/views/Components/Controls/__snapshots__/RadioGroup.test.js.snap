// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton default with empth props 1`] = `
.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  background-color: #ffffff;
}

<div
  class="c0"
/>
`;

exports[`RadioButton default with empth props 2`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c2 {
  box-sizing: border-box;
  padding: 0;
  position: absolute;
  z-index: -1;
  opacity: 0;
  overflow: visible;
}

.c3 {
  display: inline-block;
  box-sizing: border-box;
  color: #5a6169;
  width: 1.125rem;
  border-radius: 50%;
  height: 1.125rem;
  background-color: #ffffff;
  border: 1px solid #e1e5eb;
  -webkit-transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  transition: all 250ms cubic-bezier(0.27,0.01,0.38,1.06);
  box-shadow: none;
}

input:checked + .c3 {
  border: 5px solid #007bff;
}

input[type='radio'][checked][disabled] + .c3 {
  background-color: #5a6169;
  border: 5px solid #e9ecef;
}

input:disabled + .c3 {
  background-color: #e9ecef;
}

.c4 {
  margin-left: 5px;
  color: #5a6169;
}

input:disabled ~ .c4 {
  opacity: 0.7;
  color: #5a6169;
}

.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  background-color: #ffffff;
}

<div
  class="c0"
>
  <label
    checked=""
    class="c1"
  >
    <input
      checked=""
      class="c2"
      type="radio"
      value="value1"
    />
    <span
      class="c3"
    />
    <span
      class="c4"
    >
      Title1
    </span>
  </label>
  <label
    checked=""
    class="c1"
  >
    <input
      checked=""
      class="c2"
      type="radio"
      value="value2"
    />
    <span
      class="c3"
    />
    <span
      class="c4"
    >
      Title2
    </span>
  </label>
</div>
`;
